src/jit_no_ir.h

```diff
@@ -81,32 +81,134 @@ class EmitX64 {
         u16 pad{};
     };
 
+    enum class KeyPart : u32 {
+        Mod1,
+        Mod0,
+        Mod2,
+        Arp0, Arp1, Arp2, Arp3,
+        Ar0, Ar1, Pad0,
+        Mod1b,
+        Mod0b,
+        Mod2b,
+        Arp0b, Arp1b, Arp2b, Arp3b,
+        Ar0b, Ar1b, Pad1,
+        Cfgi, Cfgj,
+        Stepi0, Stepj0,
+        Cfgib, Cfgjb,
+        Stepi0b, Stepj0b,
+    };
+
+    union KeyMask {
+        std::array<u16, 32> words{};
```

Comment:
This feels a bit scary and I'm not sure if it's 100% standard-legal, might be better to use a constructor?